astropy/modeling/functional_models.py

```diff
@@ -2062,3 +2078,87 @@ def evaluate(cls, x, y, amplitude, r_eff, n, x_0, y_0, ellip, theta):
         z = np.sqrt((x_maj / a) ** 2 + (x_min / b) ** 2)
 
         return amplitude * np.exp(-bn * (z ** (1 / n) - 1))
+
+
+class BlackBody1D(Fittable1DModel):
+    """
+    One dimensional blackbody model.
+
+    Parameters
+    ----------
+    temperature : `~astropy.units.Quantity`
+        Blackbody temperature
+    bolometric_flux : `~astropy.units.Quantity`
+        The bolometric flux of the blackbody (i.e. the integral over the
+        spectral axis)
+    """
+
+    # We parametrize this model with a temperature and a bolometric flux. The
+    # bolometric flux is the integral of the model over the spectral axis. This
+    # is more useful than simply having an amplitude parameter.
+    temperature = Parameter(default=5000 * u.K, min=0)
+    bolometric_flux = Parameter(default=1 * u.erg / u.cm ** 2 / u.s)
```

Comment:
I don't get this one. Is "bolometric_flux" really necessary? What if I don't want any normalization?

My original idea is just to deprecate the blackbody functions in `analytic_functions` altogether (and in that process, get rid of that sub-package) and make this model do the calculations instead. The only reason we have `analytic_functions` at all was because we could not put blackbody in `modeling`. Now that we can, there is no reason to have duplicate functionality.